devicetypes/smartthings/viconics-schneider-room-controller.src/viconics-schneider-room-controller.groovy

```diff
@@ -0,0 +1,532 @@
+                    eventMap.name = "fanSpeed"
+                    eventMap.value = sliderValue
+                    result << createEvent([name:"thermostatFanMode", value: "on", data: [supportedThermostatFanModes: state.supportedFanModes]])
+                } else {
+                    result << createEvent([name:"thermostatFanMode", value: "auto", data:[supportedThermostatFanModes: state.supportedFanModes]])
+                }
+                break
+            case CUSTOM_THERMOSTAT_MODE:
+                log.debug "CUSTOM THERMOSTAT MODE, descMap.value: ${descMap.value}"
+                eventMap.name = "thermostatMode"
+                eventMap.value = THERMOSTAT_MODE_MAP[descMap.value]
+                eventMap.data = [supportedThermostatModes: state.supportedThermostatModes]
+                break
+            case CUSTOM_THERMOSTAT_OPERATING_STATE:
+                log.debug "CUSTOM THERMOSTAT OPERATING STATE, descMap.value: ${descMap.value}"
+                eventMap.name = "thermostatOperatingState"
+                eventMap.value = THERMOSTAT_OPERATING_STATE_MAP[descMap.value]
+                break
+            default:
+                log.debug "UNHANDLED ATTRIBUTE, descMap.inspect(): ${descMap.inspect()}"
+
+        }
+    }
+	result << createEvent(eventMap)
+	//log.debug "Description ${description} parsed to ${result}"
+	result
+}
+
+private sendEventToChild(childNumber, event) {
+	def child = childDevices?.find { getChildId(it.deviceNetworkId) == childNumber }
+
+	if (child) {
+		log.debug "Sending ${event.name} event to $child.displayName"
+		child?.sendEvent(event)
+	} else {
+		log.debug "Child device $childNumber not found!"
+	}
+}
+
+def setCoolingSetpoint(degrees) {
+	setSetpoint(degrees, COOLING_SETPOINT)
+}
+
+def setHeatingSetpoint(degrees) {
+	setSetpoint(degrees, HEATING_SETPOINT)
+}
+
+def setChildCoolingSetpoint(deviceNetworkId, degrees) {
+	log.debug "deviceNetworkId: ${deviceNetworkId} degrees: ${degrees}"
+	def switchId = getChildId(deviceNetworkId)
+	if (switchId != null) {
+		setSetpoint(degrees, COOLING_SETPOINT_UNOCCUPIED)
+	}
+}
+
+def setChildHeatingSetpoint(deviceNetworkId, degrees) {
+	log.debug "deviceNetworkId: ${deviceNetworkId} degrees: ${degrees}"
+
+	def switchId = getChildId(deviceNetworkId)
+	if (switchId != null) {
+		setSetpoint(degrees, HEATING_SETPOINT_UNOCCUPIED)
+	}
+}
+
+def getChildId(deviceNetworkId) {
+	def split = deviceNetworkId?.split(":")
+	(split.length > 1) ? split[1] as Integer : null
+}
+
+def setSetpoint(degrees, setpointAttr) {
+	log.debug "degrees: ${degrees}, setpointAttr: ${setpointAttr}"
+	if (degrees != null && setpointAttr != null) {
+		log.debug "temperatureScale: ${temperatureScale}"
+		def celsius = (temperatureScale == "C") ? degrees : fahrenheitToCelsius(degrees)
+		celsius = (celsius as Double).round(2)
+
+		delayBetween([
+				zigbee.writeAttribute(THERMOSTAT_CLUSTER, setpointAttr, DataType.INT16, zigbee.convertToHexString(celsius * 100)),
```

Comment:
Let's bring this indenting to the left one so there isn't so much wasted space

Reply:
fixed